### DISCUSSION TITLE
Override PR limit

### DISCUSSION THREAD

**Discussion by scinos** [renovatebot/renovate#15619] [Request Help] [answered]
I have a repo running local Renovate with this config (edited):
```
{
  "packageRules": [
      {
        "groupName": "Critical dependencies",
        "matchPackageNames": [
          "dep1",
          "dep2"
        ],
        "prPriority": 10
      }
  ],
  "prHourlyLimit": 3,
  "prConcurrentLimit": 3,
}
```

This works fine until there are 3 open PRs for other packages that are not in my group "Critical dependencies". At that point, a PR for "Critical dependencies" won't be created until I close one of the 3 open PRs.

Is there a way for my group "Critical dependencies" to override the limit? I'd like it to always open a PR for "Critical dependencies", even if there are 3 open PRS already.

**Comment by viceice** [comment, marked as answer]
no

**Reply by karfau** [reply]
That's it?
I have a very similar use case where I want to "exclude" certain packages from the concurrent/hourly limits, e.g. by adding the related `prConcurrentLimit` to the package rule and setting it to 0.

or even a new dedicated prop like `"excludeFromPrLimits":true`?

Is this something that we should add as a feature request, or is the renovate team saying "We will never add this option."?

**Reply by scinos** [reply]
In case it helps, we are using two renovate instances with two separate configs as a workaround: a regular instance with hourly limits, and a "expedite" instances for some selected packages with no limits.

But I'd love to see this as a feature. I had a quick look at the code and it's not trivial to implement IMO

**Comment by espada-edalex** [comment]
Just been searching for something like this too. I think ideally I'd be nice if the PR limit could be set per package manager. I was looking to see if I could set `npm` to only allow 3 MRs, `docker` to `0` (unlimited), `maven` to `5`, etc.. At the moment if I have a limit set and all the MRs are all opened by one package manager, no other MRs will open.

**Comment by MattMencel** [comment]
This is an old thread, but I just was thinking about this same question. The partial solution I've done is to increase the `prConcurrentLimit` a bit and then use `prPriority: 5` for the packageRules I want to see most frequently. Mostly for helm charts and GitHub Actions updates.

``` json
    {
      "groupName": "actions - azure",
      "commitMessageTopic": "GitHub Actions - Azure",
      "matchManagers": [
        "github-actions"
      ],
      "matchPackagePrefixes": [
        "azure/"
      ],
      "prPriority": 5
    },
 ```

**Comment by espada-edalex** [comment]
For now, I'm just reviewing the list of package bumps and manually forcing the bot to create more of them for the ones I think will be fine.

If I get time at some point, I might look into the renovate code and see if it is something that I can send in an MR for (a bit of a long stretch, but :crossed_fingers:).

**Comment by fabb** [comment]
Would be great to be able to override the limit per package rule, so i can always create PRs for dependencies that i want to `automerge`
